frontend/ui/widget/keyvaluepage.lua

```diff
@@ -184,6 +184,9 @@ function KeyValueItem:init()
             self.show_value = self.value
         end
     else
+        if self.value_overflow_align == "right" then
+            key_w = frame_internal_width - value_w_rendered
+        end
```

Comment:
I think you should have another setting named simply `value_align`, that applies if overflow or not.
(Here, you put `if value_overflow_align` in the `else` section dealing with case when there is no overflow - key and value both fit in their allowed 50%.)
So, I guess you would have here: `value_overflow_align = "right"` and `value_align = "right"`

Reply:
This is a much better solution.